lib/src/main/java/com/dynatrace/metric/util/Normalize.java

```diff
@@ -136,7 +150,39 @@ static String dimensionValue(String value) {
     return value;
   }
 
+  /**
+   * Fast check: Does the dimension value need to be escaped at all?
+   *
+   * @param dimensionValue The dimension value
+   * @return True if it contains characters that need to be escaped according to the Dynatrace
+   *     metrics ingestion protocol, false otherwise.
+   */
+  static boolean needToEscapeDimensionValue(String dimensionValue) {
+    int len = dimensionValue.length();
+    if (len > dv_max_length) {
+      return true;
+    }
+
+    for (int i = 0; i < len; i++) {
+      char c = dimensionValue.charAt(i);
+      if (c == dv_characterToEscape_quote
```

Comment:
Hm, good question. It seems as if we don't take that into account at all (dimension value characters are always escaped here, independent of whether or nor not they are in quotes). I guess we could check whether the first and last character are `"` and then adapt the check accordingly. Additionally, we would then also have to change the normalization logic for quoted strings. Or we quote all strings / we trim quotes from all strings but I don't like either of these. Either way, this increases complexity, but I think it also increases correctness, since currently some of the characters might be incorrectly escaped.

Reply:
Looking into it, it seems as if we just assume strings are unquoted (since they are always created inside the application, and somebody would have to explicitly put `String s = "\"mystring\"";` in order to even get a string like that. Of course its possible, but I think it should be very rare. I would assume that strings are generally unquoted when passed to the builder. If you pass them as quoted strings (like the example above) you probably want the escaping to be kept as is, so I would say we leave it as is and escape all characters as if they are unquoted strings.

Reply:
Agree on leaving it as is for current PR: The introduced fast check does not change the behavior of escaping as it was before it only avoids replacement in case it is not necessary (and would previously also not have been done). But true, it might warrant a follow up change (although I also agree on the unlikeliness as stated in @pirgeo's assessment).